avocado/utils/pci.py

```diff
@@ -0,0 +1,325 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Narasimhan V <sim@linux.vnet.ibm.com>
+#
+# Author: Kleber Sacilotto de Souza <klebers@linux.vnet.ibm.com>
+# Author: Daniel Kreling <kreling@linux.vnet.ibm.com>
+# for get_memory_address() and get_mask()
+
+"""
+Module for all PCI devices related functions.
+"""
+
+
+import re
+import os
+from genio import read_file
+from . import process
+
+
+def get_domains():
+    """
+    Gets all pci domains.
+    Example, it returns ['0000', '0001', ...]
+
+    :return: List of pci domains.
+    """
+    cmd = "lspci -D"
+    output = process.system_output(cmd, ignore_status=True)
+    if output:
+        domains = []
+        for line in output.splitlines():
+            domains.append(line.split(":")[0])
+        return list(set(domains))
+
+
+def get_pci_addresses():
+    """
+    Gets list of pci addresses in the system.
+    Does not return the PCI Bridges/Switches.
+
+    :return: list of pci addresses.
+    """
+    addresses = []
+    cmd = "lspci -D"
+    for line in process.system_output(cmd).splitlines():
+        if "PCI bridge" not in line:
+            addresses.append(line.split()[0])
+    if addresses:
+        return addresses
+
+
+def get_num_devices_in_domain(domain):
+    """
+    Gets number of devices in a pci domain.
+
+    :parm domain: pci domain.
+
+    :return: number of devices in a pci domain.
+    """
+    cmd = "ls -l /sys/class/*/ -1"
+    output = process.system_output(cmd, ignore_status=True)
+    if output:
+        domain = '/%s/' % domain
```

Comment:
Well this will only work if we are looking for devices inside a full pci address `0000:00:1f.2`. There are no dirs/links to only domains, so either the function name/param should be `get_num_devices_in_device` which sounds strange (I think more fitting would be `get_num_interfaces_in_device`). Alternatively you really want to get num of interfaces from all devices on a given bus, then you have to use the `domain = "/%s" % domain` to get everything which starts with the domain, but it will be prone to false matches (on the other hand it'd allow searching for either domains, domain+bus, domain+bus+device or full domain+bus+device+function).

I'd suggest adding an example to each (or at least the most problematic) params in docstrings to avoid such confusions.